site/app/templates/admin/admin_gradeable/AdminGradeablePeers.twig

```diff
@@ -169,8 +183,134 @@
         });
     }
 
+    function addGraderPeerForm() {
+        // make add grader popup appear
+        $('.popup-form').css('display', 'none');
+        let form = $("#admin-gradeable-peers-form2");
+        form.css("display", "block");
+        $('#admin-gradeable-peers2-submit').prop('disabled', true);
+        captureTabInModal("admin-gradeable-peers-form2");
+        // set the left panel to allow a new grader to be added
+        let peer_grader_pairs = JSON.parse($('#peer_grader_pairs').val());
+        let add_grader_div = $("#new-grader");
+        add_grader_div.empty();
+        add_grader_div.append("Grader to Add:<br />")
+        add_grader_div.append('<input type="text" id="new_peer_grader" name="new_peer_grader" /><br>');
+        let student_autofill = JSON.parse($('#student_autofill').val());
+        $('[name="new_peer_grader"]', form).autocomplete({
+            source: student_autofill
+        });
+        $('[name="new_peer_grader"]').autocomplete( "option", "appendTo", form ); 
+        // set the right panel to allow adding students for the new grader
+        let grader_students_div = $("#new-grader-students");
+        grader_students_div.empty();
+        grader_students_div.append("Students for New Grader:<br />")
+        grader_students_div.append('<input class="add_user_ids" type="text" name="add_user_id_0" /><br>');
+        $('[name="add_user_id_0"]', form).autocomplete({
+            source: student_autofill
+        });
+        $('[name="add_user_id_0"]').autocomplete( "option", "appendTo", form );
+        grader_students_div.append('<p id="add_student_button" style="cursor: pointer;" onclick="addStudentTextbox(\'#admin-gradeable-peers-form2\',\'#new-grader-students\');" aria-label="Add More Users"><i class="fas fa-plus-square"></i>Add More Users<br></p>');
+    }
+
+    function checkNewGraderValid(user_input) {
+        // check if the new grader is a valid student by autofill
+        let student_autofill = JSON.parse($('#student_autofill').val());
```

Comment:
is the value from the student_autofill guaranteed to be valid JSON? If there's not this should be in a try catch incase JSON parse throws an exception.